src/engine/test/source/builder/builders/opBuilderKVDBSet_test.cpp

```diff
@@ -0,0 +1,250 @@
+    expectedEvent->setBool(true, "/output");
+
+    string dbOp1 {DB_NAME};
+    string keyOp1 {"some_key"};
+    string valueOp1 {"some_value"};
+    auto parameters = std::make_tuple<string, string, vector<string>>("/output", "", {DB_NAME, keyOp1, valueOp1});
+    const auto op1 = getOpBuilderKVDBSet(kvdbManager)(parameters);
+
+    string dbOp2 {DB_NAME};
+    string keyOp2 {"fieldString"};
+    string valueOp2 {"$fieldString"};
+    parameters = std::make_tuple<string, string, vector<string>>("/output", "", {DB_NAME, keyOp2, valueOp2});
+    const auto op2 = getOpBuilderKVDBSet(kvdbManager)(parameters);
+
+    string dbOp3 {DB_NAME};
+    string keyOp3 {"fieldIntNumber"};
+    string valueOp3 {"$fieldIntNumber"};
+    parameters = std::make_tuple<string, string, vector<string>>("/output", "", {DB_NAME, keyOp3, valueOp3});
+    const auto op3 = getOpBuilderKVDBSet(kvdbManager)(parameters);
+
+    string dbOp4 {DB_NAME};
+    string keyOp4 {"fieldObject"};
+    string valueOp4 {"$fieldObject"};
+    parameters = std::make_tuple<string, string, vector<string>>("/output", "", {DB_NAME, keyOp4, valueOp4});
+    const auto op4 = getOpBuilderKVDBSet(kvdbManager)(parameters);
+
+    string dbOp5 {DB_NAME};
+    string keyOp5 {"fieldArray"};
+    string valueOp5 {"$fieldArray"};
+    parameters = std::make_tuple<string, string, vector<string>>("/output", "", {DB_NAME, keyOp5, valueOp5});
+    const auto op5 = getOpBuilderKVDBSet(kvdbManager)(parameters);
+
+    string dbOp6 {DB_NAME};
+    string keyOp6 {"fieldNull"};
+    string valueOp6 {"$fieldNull"};
+    parameters = std::make_tuple<string, string, vector<string>>("/output", "", {DB_NAME, keyOp6, valueOp6});
+    const auto op6 = getOpBuilderKVDBSet(kvdbManager)(parameters);
+
+    string dbOp7 {DB_NAME};
+    string keyOp7 {"fieldDoubleNumber"};
+    string valueOp7 {"$fieldDoubleNumber"};
+    parameters = std::make_tuple<string, string, vector<string>>("/output", "", {DB_NAME, keyOp7, valueOp7});
+    const auto op7 = getOpBuilderKVDBSet(kvdbManager)(parameters);
+
+    string dbOp8 {DB_NAME};
+    string keyOp8 {"fieldTrue"};
+    string valueOp8 {"$fieldTrue"};
+    parameters = std::make_tuple<string, string, vector<string>>("/output", "", {DB_NAME, keyOp8, valueOp8});
+    const auto op8 = getOpBuilderKVDBSet(kvdbManager)(parameters);
+
+    string dbOp9 {DB_NAME};
+    string keyOp9 {"fieldFalse"};
+    string valueOp9 {"$fieldFalse"};
+    parameters = std::make_tuple<string, string, vector<string>>("/output", "", {DB_NAME, keyOp9, valueOp9});
+    const auto op9 = getOpBuilderKVDBSet(kvdbManager)(parameters);
+
+    auto result = op1->getPtr<Term<EngineOp>>()->getFn()(event);
+    ASSERT_TRUE(result);
+    ASSERT_EQ(*result.payload(), *expectedEvent);
+    auto rawValue = kvdbManager->getRawValue(DB_NAME, keyOp1);
+    if (const auto err = std::get_if<base::Error>(&rawValue))
+    {
+        throw std::runtime_error(err->message);
+    }
```

Comment:
can we merge this into an  `ASSERT_THROW`  ?

Reply:
`base::Error` is not an exception but a message, so no, it cannot be catch with an `ASSERT_THROW`